docs/tutorials/discord-slash-command.md

```diff
@@ -0,0 +1,149 @@
+
+| Name             | Value                                                                                       |
+| ---------------- | ------------------------------------------------------------------------------------------- |
+| discord_token    | Copied from Discord Developer Portal                                                        |
+| bot_id | Copied from Discord Developer Portal -- General information -- Application ID  |
+
+Click on the **Build** button to create the flow function.
+
+In the next screen, you will be asked to connect to Discord. Since we are providing a Discord API token to access the bot here, you can simply click on **Deploy**.
+
+## Deploy
+
+Finally, you will be redirected to the flow details page, where you can check for service status and logs. Discord and OpenAI should appear as connected external services on this page.
+
+When the status of the flow is ready and running, you can invite the Discord bot to your server.
+
+> Refer to this guide to [invite the bot to your server](https://flows.network/blog/discord-chat-bot-guide).
+
+After the bot joins your server, you can find the bot on the right online contact list. Then you can go to the specific channel and type /weather + name to give it a try.
+
+## Code walkthrough
+
+The source code for the flow function is written in the Rust programming language.  Let's start with how to register a slash command for a bot. The following code registers a slash command `weather` for a Discord bot with the `register_commands` function using Discord's HTTP API.
+
+```
+async fn register_commands() {
+
+    let bot_id = env::var("bot_id").unwrap_or("1124137839601406013".to_string());
+ // the details of a slash command
+    let command = serde_json::json!({
+        "name": "weather",
+        "description": "Get the weather for a city",
+        "options": [
+            {
+                "name": "city",
+                "description": "The city to lookup",
+                "type": 3,
+                "required": true
+            }
+        ]
+    });
+
+    let discord_token = env::var("discord_token").unwrap();
+    let http_client = HttpBuilder::new(discord_token)
+        .application_id(bot_id.parse().unwrap())
+        .build();
+
+    match http_client
+        .create_global_application_command(&command)
+        .await
+    {
+        Ok(_) => log::info!("Successfully registered command"),
+        Err(err) => log::error!("Error registering command: {}", err),
+    }
+}
+```
+Now we have registered a slash command `weather` for the discord bot. Next, let's see how to listen for a slash command. The `handler` function is annotated with `#[application_command_handler]`. If the input matches a `/weather` command, it will extract the `city` option from the command arguments and then call the `get_weather()` function to look up weather data for the given city. Finally, it will send the content to the channel. 
```

Comment:
"Now we have registered a slash command. It is shown on the Discord UI when the user tries to send a message to the bot. When the user sends a slash command to the bot, we need to capture and handle it in our application."